xrpl/models/transactions/signer_list_set.py

```diff
@@ -56,6 +56,23 @@ def is_dict_of_model(cls: Type[SignerEntry], dictionary: Dict[str, Any]) -> bool
             and super().is_dict_of_model(dictionary["signer_entry"])
         )
 
+    @classmethod
+    def from_dict(cls: Type[SignerEntry], value: Dict[str, Any]) -> SignerEntry:
+        """
+        Construct a new SignerEntry from a dictionary of parameters.
+
+        Args:
+            value: The value to construct the SignerEntry from.
+
+        Returns:
+            A new SignerEntry object, constructed using the given parameters.
+        """
+        if len(value) == 1 and "signer_entry" in value:
+            return cast(
+                SignerEntry, super(SignerEntry, cls).from_dict(value["signer_entry"])
+            )
+        return cast(SignerEntry, super(SignerEntry, cls).from_dict(value))
```

Comment:
i'm thinking that all this mussing of types and values is a good sign we don't have the optimal design. this solution we have is too complicated for the problem.

but again - for a later release

Reply:
Sorry, I don't follow - do you mean the casting?